crates/notion-core/src/inventory/serial.rs

```diff
@@ -161,3 +181,156 @@ impl YarnIndex {
         Ok(super::YarnIndex { entries })
     }
 }
+
+// see npm registry API doc:
+// https://github.com/npm/registry/blob/master/docs/REGISTRY-API.md
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct PackageMetadata {
+    pub name: String,
+    pub description: Option<String>,
+    pub versions: HashMap<String, PackageVersionInfo>,
+    #[serde(rename = "dist-tags")]
+    pub dist_tags: PackageDistTags,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct PackageVersionInfo {
+    // there's a lot more in there, but right now just care about the version
+    pub version: String,
+    pub dist: DistInfo,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct PackageDistTags {
+    pub latest: String,
+    pub beta: Option<String>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct DistInfo {
+    pub shasum: String,
+    pub tarball: String,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct PackageConfig {
+    pub name: String,
+    pub version: String,
+    pub platform: toolchain::serial::Platform,
+    pub bins: Vec<String>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct BinConfig {
+    pub name: String,
+    pub package: String,
+    pub version: String,
+    pub path: String,
+    pub platform: toolchain::serial::Platform,
+}
+
+impl PackageMetadata {
+    pub fn into_index(self) -> Fallible<package::PackageIndex> {
+        let latest_version = Version::parse(&self.dist_tags.latest).unknown()?;
+
+        let mut entries = Vec::new();
+        for (_, version_info) in self.versions {
+            let parsed_version = Version::parse(&version_info.version).unknown()?;
```

Comment:
Hah, good catch! I agree that it should be a `Version` in the deserializer. I think I originally had this as a `String` before the semver parsing was updated, and then never updated it - going to do that now.

Reply:
Oh actually this is the package metadata - not sure why I have this as a string then. Anyway I'm going to change it.

Reply:
Ah I see why - because deserializing `Version` doesn't play nice with serde out of the box:
```
the trait bound `semver::version::Version: event::_IMPL_SERIALIZE_FOR_Event::_serde::Deserialize<'_>` is not satisfied
```

I'll have to hook that in as the deserializer

Reply:
Ok that's changed now, all of those fields use `Version` instead of `String`